### PR TITLE
WINDUP-612  Validate all XML rules in windup-rulesets during build.

### PR DESCRIPTION
**Pull request by OndraZizka**
Depends on https://github.com/windup/windup/pull/570

### CODE REVIEW

**Comment by lincolnthree** [comment]
I think this is the wrong solution. I'd rather point the build to a hosted schema than pull it out of an artifact like this. Let's think about other solutions and discuss before implementing this.

**Comment by OndraZizka** [comment]
But on meeting, you said that the authoritative place will be github repository, probably config-xml. This is the implementation of it.

**Comment by mareknovotny** [comment]
We don't bundle schema into a jar/artifact for now, right? It would make sense to bundle it and validate to it from already downloaded Maven artifact

**Comment by OndraZizka** [comment]
New PR to bundle the XSD in windup core:
https://github.com/windup/windup/pull/582

File: pom.xml [hunk 1]
```diff
@@ -167,5 +167,50 @@
...
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>xml-maven-plugin</artifactId>
+                <version>1.0</version>
+                <executions> <execution> <goals> <goal>validate</goal> </goals> </execution> </executions>
```
Comment (lincolnthree) [inline pom.xml]: Formatting. Newlines needed here.

File: pom.xml [hunk 2]
```diff
@@ -167,5 +167,50 @@
...
+                <configuration>
+                    <validationSets>
+                        <validationSet>
+                            <includes>
+                                <include>**/*.windup.xml</include>
+                            </includes>
+                            <dir>rules</dir>
+                            <systemId>${project.build.directory}/xsd/windup-ruleset-2.3.0.Final.xsd</systemId>
```
Comment (lincolnthree) [inline pom.xml]: This should reference the version-less schema.

File: pom.xml (hunk 1 above)
Comment (OndraZizka) [inline pom.xml]: Changed. I will never understand why they are needed, though.

File: pom.xml (hunk 2 above)
Comment (OndraZizka) [inline pom.xml]: Fixed.

**Comment by OndraZizka** [comment]
Seems like I need to re-create the PR, github doesn't let me reopen.

**Pull request closed without merging**